contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.5.0--4.6.0.sql

```diff
@@ -1041,6 +1041,203 @@ CREATE OR REPLACE FUNCTION sys.json_query(json_string text, path text default '$
+        WHERE n.nspname = 'sys' 
+        AND p.proname = 'babelfish_conv_helper_to_varbinary'
+        AND p.pronargs = 3  -- old version with 3 parameters
+        AND p.proargtypes[0] = 'anyelement'::regtype::oid
+    ) INTO old_function_exists;
+
+    IF old_function_exists THEN
+        -- Recreate definition with updated dependant function syntax.
+        CREATE OR REPLACE FUNCTION sys.babelfish_conv_helper_to_varbinary(
+            IN arg anyelement,
+            IN try BOOL,
+            IN p_style NUMERIC DEFAULT 0
+        )
+        RETURNS sys.varbinary
+        AS
+        $BODY$
+        DECLARE result sys.varbinary;
+        BEGIN
+            IF try THEN
+                --  Hardcoding this as the internal function could have been dropped)
+                RETURN sys.babelfish_try_conv_to_varbinary(-1 , arg, p_style);
+            ELSE
+                IF pg_typeof(arg) IN ('text'::regtype, 'sys.ntext'::regtype, 'sys.nvarchar'::regtype, 'sys.bpchar'::regtype, 'sys.nchar'::regtype) THEN
+                    RETURN sys.babelfish_conv_string_to_varbinary(arg, p_style);
+                ELSE
+                    RETURN CAST(arg as sys.varbinary);
+                END IF;
+            END IF;
+        END;
+        $BODY$
+        LANGUAGE plpgsql
+        IMMUTABLE;
+
+        ALTER FUNCTION sys.babelfish_conv_helper_to_varbinary(
+            IN arg anyelement,
+            IN try BOOL,
+            IN p_style NUMERIC
+        ) RENAME TO babelfish_conv_helper_to_varbinary_anyel_deprecated_in_5_2_0;
+
+        CREATE OR REPLACE FUNCTION sys.babelfish_conv_helper_to_varbinary(
+            IN typmod INTEGER,
+            IN arg anyelement,
+            IN try BOOL,
+            IN p_style NUMERIC DEFAULT 0
+        )
+        RETURNS sys.varbinary
+        AS
+        $BODY$
+        DECLARE result sys.varbinary;
+        BEGIN
+            IF try THEN
+                RETURN sys.babelfish_try_conv_to_varbinary(typmod, arg, p_style);
+            ELSE
+                IF pg_typeof(arg) IN ('text'::regtype, 'sys.ntext'::regtype, 'sys.nvarchar'::regtype, 'sys.bpchar'::regtype, 'sys.nchar'::regtype) THEN
+                    RETURN sys.babelfish_conv_string_to_varbinary(arg, p_style);
+                ELSE
+                    IF typmod = -1 THEN
+                        RETURN CAST(arg as sys.varbinary);
+                    ELSE
+                        -- Pass typmod-4 (Way typmod for sys.binary/varbinary is stored) to helper functions to use the typmod in initial conversion
+                        EXECUTE format('SELECT CAST($1 as sys.varbinary(%s))', typmod - 4) INTO result USING arg;
+                        RETURN result;
+                    END IF;
+                END IF;
+            END IF;
+        END;
+        $BODY$
+        LANGUAGE plpgsql
+        IMMUTABLE;
+
+        CALL sys.babelfish_drop_deprecated_object('function', 'sys', 'babelfish_conv_helper_to_varbinary_anyel_deprecated_in_5_2_0');
```

Comment:
[Question] What if a cx view depends upon this function so we won't be able to drop it but we successfully re-created `babelfish_try_conv_to_varbinary`, in that case the old view will this deprecated function and it will call `babelfish_try_conv_to_varbinary` with two arguments but it won't exist.
IMO, we should DROP/deprecate all once instead of one at a time to avoid this.

Reply:
I have recreated the function to use modern signature before renaming it to deprecated to use the newer version
[link](https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/3597/files#diff-864f9aa8e9ecd92caf82f79ca1014f007d34c1420834c1542051be9d5db207b9R1167-R1192)